tdrs-frontend/src/components/PermissionGuard/PermissionGuard.jsx

```diff
@@ -0,0 +1,47 @@
+import { useSelector } from 'react-redux'
+import {
+  accountStatusIsApproved,
+  selectUserPermissions,
+} from '../../selectors/auth'
+
+const isAllowed = (
+  { permissions, isApproved },
+  requiredPermissions,
+  requiresApproval
+) => {
+  if (requiresApproval && !isApproved) {
+    return false
+  }
+
+  if (!requiredPermissions) {
+    return true
+  }
+
+  for (var i = 0; i < requiredPermissions.length; i++) {
+    if (!permissions.includes(requiredPermissions[i])) {
+      return false
```

Comment:
Something about this gives me a code smell but it is short, readable, and I imagine easy to troubleshoot in the future compared to something like `map`/`reduce`/`for..in`. Minor nitpick and don't believe it's worth refactoring.

Reply:
javascript doesn't have a `foreach`, just `Array.prototype.forEach` which accepts a function `(e, i, a)` that runs for each element. that means we can't `return` to break the loop and exit the function, we'd have to keep a `let isAllowed = true` outside the loop. either is fine, i don't think one way is necessarily cleaner than the other. but yeah, that's why it's like this :D